llvm/lib/CodeGen/MachineOutliner.cpp

```diff
@@ -684,7 +688,7 @@ void MachineOutliner::findCandidates(
       continue;
     }
 
-    FunctionList.push_back(*OF);
+    FunctionList.push_back(std::make_unique<OutlinedFunction>(*OF));
```

Comment:
Does this work?
```suggestion
    FunctionList.emplace_back(*OF);
```

Reply:
This isn't quite right, but either case created implicit copy from the local OutlinedFunction instance. Instead, I also make it unique when it's created in `getOutliningCandidateInfo` so that we can simply move it here.